Tiltfile

```diff
@@ -396,6 +401,77 @@ def prepare_all():
     )
     local(cmd, env = settings.get("kustomize_substitutions", {}))
 
+# create flavor resources from cluster-template files in the templates directory
+def flavors():
+    substitutions = settings.get("kustomize_substitutions", {})
+
+    template_list = [item for item in listdir("./test/infrastructure/docker/templates")]
+    template_list = [template for template in template_list if os.path.basename(template).endswith("yaml")]
+
+    for template in template_list:
+        deploy_worker_templates(template, substitutions)
+
+    local_resource(
+        name = "delete-all-workload-clusters",
+        cmd = kubectl_cmd + " delete clusters --all --wait=false",
+        auto_init = False,
+        trigger_mode = TRIGGER_MODE_MANUAL,
+        labels = ["flavors"],
+    )
+
+def deploy_worker_templates(template, substitutions):
+    # validate template exists
+    if not os.path.exists(template):
+        fail(template + " not found")
+
+    yaml = str(read_file(template))
+    flavor = os.path.basename(template).replace("cluster-template-", "").replace(".yaml", "")
+
+    # for the base cluster-template, flavor is "default"
+    flavor = os.path.basename(flavor).replace("cluster-template", "default")
```

Comment:
I think in either case it would be beneficial to support ClusterClass. I'm not familiar with using ClusterClasses yet though. Is the basic use case that given a cluster class yaml file, we want to run `clusterctl generate cluster` and use `kubectl apply -f` on the output?